### DISCUSSION TITLE
How to edit only parts of states

### DISCUSSION THREAD

**Discussion by THEkatinamicrowave** [CodenameCrew/CodenameEngine#461] [Q&A] [answered]
I want to change the healthicon position and functionality, but that would involve redirecting PlayState to a ModdedPlayState which would totally break the mod whenever a new action build comes out
How do I only edit that

**Comment by r6915ee** [comment, marked as answer]
> I want to change the healthicon position and functionality, but that would involve redirecting PlayState to a ModdedPlayState which would totally break the mod whenever a new action build comes out How do I only edit that

You don't really need to create a modded version of `PlayState`. Important `PlayState` variables are available from any song script, and those include the icon variables (`iconBF` and `iconDad`, for the player and opponent respectively). As a result, you can modify the icon variables as you normally would in any script loaded by the game during normal gameplay.

**Reply by FuroYT** [reply]
For clarification iconBF and iconDad does not exists and to access the icons just use iconP1 and iconP2

**Reply by THEkatinamicrowave** [reply]
yeah I know what he meant
all public vars can be accessed by scripts
I knew that was a thing but I didn't want to put that script in every 1-player song in my mod so I wanted to mod PlayState or smt it's fine though that idea was horrible from the getgo

**Reply by FuroYT** [reply]
You can have a global song script for that if you use a script in the songs folder of your mod

**Reply by THEkatinamicrowave** [reply]
oh ok thanks

**Reply by r6915ee** [reply]
> For clarification iconBF and iconDad does not exists and to access the icons just use iconP1 and iconP2

Wait, is that the case for newer versions of Codename? I've seen some implementations of custom HUDs use iconBF and iconDad.

**Reply by SenDoesStuff** [reply]
Most custom HUDs make their own icons rather than using the base ones so they dont have to work around pre-existing code that would affect them in ways they dont want

**Reply by THEkatinamicrowave** [reply]
guys chill I didn't mean to start this 😭😭😭

**Reply by NexIsDumb** [reply]
> guys chill I didn't mean to start this 😭😭😭

wdym nobody is fighting lmao
